tests/integration/targets/mso_schema_site_contract_service_graph/tasks/main.yml

```diff
@@ -0,0 +1,770 @@
+    epg: EPG1
+    contract:
+      name: Contract1
+      type: provider
+    state: present
+
+- name: Bind Contract1 with EPG1 consumer
+  cisco.mso.mso_schema_template_anp_epg_contract:
+    <<: *mso_info
+    schema: "ansible_schema"
+    template: "ansible_template1"
+    anp: AP1
+    epg: EPG1
+    contract:
+      name: Contract1
+      type: consumer
+    state: present
+
+- name: Bind Contract1 with template service graph - ansible_service_graph
+  cisco.mso.mso_schema_template_contract_service_graph:
+    <<: *mso_info
+    schema: "ansible_schema"
+    template: "ansible_template1"
+    contract: Contract1
+    service_graph: ansible_service_graph
+    service_nodes:
+      - provider: BD1
+        consumer: BD2
+      - provider: BD1
+        consumer: BD2
+      - provider: BD1
+        consumer: BD2
+    state: present
+
+# # Test part
+- name: Bind the Contract1 with site service graph - ansible_service_graph - check_mode
+  cisco.mso.mso_schema_site_contract_service_graph: &add_site_con_sg_cm
+    <<: *mso_info
+    tenant: ansible_tenant
+    schema: ansible_schema
+    template: ansible_template1
+    site: ansible_test
+    contract: Contract1
+    service_graph_schema: ansible_schema
+    service_graph_template: ansible_template1
+    service_graph: ansible_service_graph
+    node_relationship:
+      - cluster_interface_device: ansible_tenant_firewall1
+        provider_connector_cluster_interface: clu_if1
+        provider_connector_redirect_policy: redirect_policy1
+        consumer_connector_cluster_interface: clu_if1
+        consumer_connector_redirect_policy: redirect_policy1
+      - cluster_interface_device: ansible_tenant_adc
+        provider_connector_cluster_interface: clu_if3
+        provider_connector_redirect_policy: redirect_policy1
+        consumer_connector_cluster_interface: clu_if3
+        consumer_connector_redirect_policy: redirect_policy1
+        consumer_subnet_ips: ["1.1.1.1/24", "4.4.4.4/24"]
+      - cluster_interface_device: ansible_tenant_other
+        provider_connector_cluster_interface: clu_if4
+        provider_connector_redirect_policy: redirect_policy1
+        consumer_connector_cluster_interface: clu_if4
+        consumer_connector_redirect_policy: redirect_policy1
+    state: present
+  check_mode: true
+  ignore_errors: true
+  register: add_sg_cm
+
+- name: Bind the Contract1 with site service graph - ansible_service_graph - normal_mode
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *add_site_con_sg_cm
+  ignore_errors: true
+  register: add_sg_nm
+
+- name: Bind the Contract1 with site service graph - ansible_service_graph - normal_mode - again
+  cisco.mso.mso_schema_site_contract_service_graph:
+    <<: *add_site_con_sg_cm
+  ignore_errors: true
+  register: add_sg_nm_again
+
+- name: Assertions check for the bind site contract service graph
+  assert:
+    that:
+      - add_sg_cm is changed
+      - add_sg_cm.current != {}
+      - add_sg_cm.previous == {}
```

Comment:
I think we should assert proposed values for checkmode. Applies to all tests assertions with checkmode